src/test/java/io/vertx/test/codegen/testapi/MethodWithInvalidTypeParamByParameterizedReturn.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.test.codegen.testapi;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.codegen.testmodel.TestDataObject;
+
+/**
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+@VertxGen
+public interface MethodWithInvalidTypeParamByParameterizedReturn {
```

Comment:
it's split because when you have a failures with an interface that contains 2 methods, you know that just one of them is wrong, not all of them.